spec/helpers/application_helper_spec.rb

```diff
@@ -62,6 +62,11 @@
   end
 
   describe "#release_adjusted_cache_key" do
+    before do
+      stub_const("ENV", ENV.to_h.merge("HEROKU_RELEASE_CREATED_AT" => ""))
```

Comment:
I don't make sense that you need to to stub `HEROKU_RELEASE_CREATED_AT` because each of the spec in this describe block is already stubbing with `allow(ApplicationConfig).to ...`

I think what you need instead is an afterblock to clear the variable created.
```
after do
   ForemInstance.instance_variable_set(:@deployed_at, nil)
end
```

Reply:
Also, if this is a caching related issue, you should be able to reproduce it consistently locally. And because of that, you should be able to solidly verify if this fix will work or not.

Reply:
@maestromac my suspicion is when the first test runs in the describe block then it has the cached value `A deploy date set on Heroku` which was set on `HEROKU_RELEASE_CREATED_AT` in another test. (

Hence I thought that I'd rather be 100% sure that the variable was clear before running the tests, as this will pretty much mimic a real scenario. It's pretty much impossible to test the change locally since this does not replicate locally. 

That being said, just setting the `ForemInstance.instance_variable_set(:@deployed_at, nil)` seemed to have fixed the issue on [another branch](https://github.com/forem/forem/pull/12292/commits), but again I didn't think there was too much harm in making sure that the Heroku Variable was also clear, but if you feel  like just resetting the ForemInstance variable is sufficient, I can undo the HEROKU_RELEASE_CREATED_AT stub.

Reply:
In addition, the reason I put `ForemInstance.instance_variable_set(:@deployed_at, nil)` in a before block instead of an after block is incase the cached variable carries over from another test, which has happened before.

It should do the same thing if I move it to after blocks in all the tests where we've added this line.

Reply:
Ah sorry i didn't read your updated your comment about the `after blocks`. What you're saying makes sense, and I can move them all to after blocks instead.

Reply:
> Also, if this is a caching related issue, you should theoretically be able to reproduce it consistently locally. And because of that, you should be able to solidly verify if this fix will work or not.

I've been grasping at straws, so would love your help if you able to provide more insight. 
I can't definitely tell what the issue is, but I do believe it's a combination of the the order the tests are running in, and the memoization. But again, there has been now way to confirm it locally. However, this fix did seem to sort out the issue in two different PR's; https://github.com/forem/forem/pull/12292/commits and https://github.com/forem/forem/pull/12262